### ISSUE TITLE
Autcomplete search is not accessible (high)

### ISSUE THREAD

**Issue by MelanieHotchkiss** [magda-io/magda#1913]
### Problem description
Intopia issue 22 - Screen reader users are not advised that the search box (the input field) has an autocomplete function. Screen reader users are not advised that they can select from a list of options or type in their own input. When a user enters a search string in the search box, the list of matching results is updated visually but this is not communicated to screen reader users. The number of matching results is also updated visually but is not communicated by screen readers.

### Problem reproduction steps
To tell screen reader users that the input field is an auto-suggest, add role=”combobox” to the <input> field markup.
To tell screen reader users they can select from the suggestion list or type their own input, add aria-autocomplete=”both” to the <input> field markup.
We recommend reviewing the potential for the application of techniques described in the article Anatomy of an Accessible Auto Suggest.

### Screenshot / Design / File reference
[22] Autcomplete search is not accessible
Type: WCAG 2.1 failure
Fails 4.1.2 Name, Role, Value - Level A
Choose a Success Criteria
Location: Search, Search page
Platform: Desktop
Severity: High
What’s the issue?
Screen reader users are not advised that the search box (the input field) has an autocomplete function. Screen reader users are not advised that they can select from a list of options or type in their own input. When a user enters a search string in the search box, the list of matching results is updated visually but this is not communicated to screen reader users. The number of matching results is also updated visually but is not communicated by screen readers.
Why it matters
Not telling screen reader users that the input field has an autocomplete function is a failure of Success Criterion 4.1.2, which requires that changes to user interface components be communicated to all users.
Dynamically updating the visual display of matching search results and not communicating that to screen reader is also a failure of Success Criterion 4.1.2.
Since the display of the number of matching results cannot take focus and is not announced, it also fails Success Criterion 4.1.2.
Autocomplete search is extremely useful but it requires the application of some specific techniques to be accessible.
intopia.digi tal 31
Our recommendation
To tell screen reader users that the input field is an auto-suggest, add role=”combobox” to the <input> field markup.
To tell screen reader users they can select from the suggestion list or type their own input, add aria-autocomplete=”both” to the <input> field markup.
We recommend reviewing the potential for the application of techniques described in the article Anatomy of an Accessible Auto Suggest.
